planetiler-core/src/main/java/com/onthegomap/planetiler/pmtiles/Pmtiles.java

```diff
@@ -324,4 +333,41 @@ public static List<Entry> deserializeDirectory(byte[] bytes) {
     }
     return result;
   }
+
+  private static final ObjectMapper objectMapper = new ObjectMapper()
+    .registerModules(new Jdk8Module())
+    .setSerializationInclusion(NON_ABSENT);
+
+  /**
+   * Arbitrary application-specific JSON metadata in the archive.
+   * <p>
+   * stores name, attribution, created_at, planetiler build SHA, vector_layers, etc.
+   */
+  public record JsonMetadata(
+    @JsonProperty("vector_layers") List<LayerStats.VectorLayer> vectorLayers,
+    @JsonAnyGetter @JsonAnySetter Map<String, String> otherMetadata
+  ) {
+
+    @JsonCreator
+    JsonMetadata(@JsonProperty("vector_layers") List<LayerStats.VectorLayer> vectorLayers) {
+      this(vectorLayers, new HashMap<>());
+    }
+
+    public byte[] toBytes() {
```

Comment:
The tests will catch that we can deserialize what we serialize, but will they confirm this actually hoists the other metadata attributes to top-level? If now you could try deserializing the bytes as a map and confirm it has keys from otherMetadata.

Reply:
https://github.com/onthegomap/planetiler/pull/502/commits/f03bc180aa5b3a873de59c19980a162413aef0e7